crates/utils/src/tracing/mod.rs

```diff
@@ -1,6 +1,73 @@
 pub mod grpc;
 
-// Re-export useful traits for open-telemetry traces. This avoids requiring other crates from
-// importing that family of crates directly.
-pub use opentelemetry::trace::Status as OtelStatus;
-pub use tracing_opentelemetry::OpenTelemetrySpanExt;
+use core::time::Duration;
+
+use miden_objects::{block::BlockNumber, Digest};
+use opentelemetry::{trace::Status as OtelStatus, Key, Value};
+use sealed::sealed;
+
+/// Utility functions for converting types into [`opentelemetry::Value`].
+pub trait ToValue {
+    fn to_value(&self) -> Value;
+}
+
+impl ToValue for Duration {
+    fn to_value(&self) -> Value {
+        self.as_secs_f64().into()
+    }
+}
+
+impl ToValue for Digest {
+    fn to_value(&self) -> Value {
+        self.to_hex().into()
+    }
+}
+
+impl ToValue for f64 {
+    fn to_value(&self) -> Value {
+        (*self).into()
+    }
+}
+
+impl ToValue for BlockNumber {
+    fn to_value(&self) -> Value {
+        i64::from(self.as_u32()).into()
+    }
+}
+
+impl ToValue for u32 {
+    fn to_value(&self) -> Value {
+        i64::from(*self).into()
+    }
+}
+
+impl ToValue for i64 {
+    fn to_value(&self) -> Value {
+        (*self).into()
+    }
+}
+
+/// Utility functions based on [`tracing_opentelemetry::OpenTelemetrySpanExt`].
+#[sealed]
+pub trait OpenTelemetrySpanExt {
+    fn set_attribute(&self, key: impl Into<Key>, value: impl ToValue);
+    fn set_error(&self, err: &dyn std::error::Error);
+}
```

Comment:
I don't think we need a crate dependency for sealing. You can do sealing as follows:

```suggestion
pub trait OpenTelemetrySpanExt: Sealed {
    fn set_attribute(&self, key: impl Into<Key>, value: impl ToValue);
    fn set_error(&self, err: &dyn std::error::Error);
}

mod private {
    pub trait Sealed {}
}
```

Since we're only implementing the trait for on type (`tracing::Span`), the proc macro doesn't really buy us much.